flink/src/main/java/org/apache/iceberg/flink/FlinkCatalog.java

```diff
@@ -447,10 +459,12 @@ private static void validateFlinkTable(CatalogBaseTable table) {
     if (!schema.getWatermarkSpecs().isEmpty()) {
       throw new UnsupportedOperationException("Creating table with watermark specs is not supported yet.");
     }
+  }
 
-    if (schema.getPrimaryKey().isPresent()) {
-      throw new UnsupportedOperationException("Creating table with primary key is not supported yet.");
-    }
+  private static List<String> toEqualityColumns(TableSchema schema) {
```

Comment:
In the format, I think that equality columns should be tracked by ID rather than by name so that renaming columns doesn't break the primary key metadata.

Reply:
Yes,  it should track ID here.  I was thinking that it could be a temporary solution before introducing the primary key   specification.  Considering the renaming issue,  it seems not a good idea to track column names in iceberg properties. 

I will create another PR to introduce the primary key specification based on the discussion from [here](https://github.com/apache/iceberg/pull/1974#discussion_r547616494).